packages/shared/src/hooks/companion/common.ts

```diff
@@ -0,0 +1,4 @@
+export const getCompanion = (): HTMLElement =>
```

Comment:
I'm not really sure why this makes sense in shared if only extension can ever use it?

Reply:
Right. Let me see how I can just pass the requirement of this one.

Reply:
There will be a lot of prop drilling to happen if we don't host it here. Another option is to another property in the `useRequestProtocol` on where to append tooltips. Though it feels out of context for the said hook by its name.

Probably a new hook that sets it across the app? Would also help a lot of other areas (mostly on modals) to not think about it anymore whenever working with tooltips/popups. WDYT?

Reply:
@rebelchris would love to hear what you think on this one.

Reply:
I personally think anything "extension" "companion" only should be in that repo, not in shared.
It shouldn't include prop drilling, but maybe I'm not getting the right picture here?

Reply:
When the app is Companion, we need to append the tooltips to the ShadowRoot in order to display the right content. We have a preview tab in the comment box where the tooltip is used when hovering a mention preview, this same shared component is used for making a comment in the companion. That means we will have to drill down the companion wrapper from the app down to the comment box because the shared component itself will append it to the default value which is the `document`.

Reply:
Well yes, but what does it matter if this code sits in "extension" or "shared"?

Reply:
So the shared can use it too since it would be then a dependency cycle if the `shared` depends on the `extension`.